### PR TITLE
move common code of reports_controller and timelog_controller into a common helper

### PR DESCRIPTION
**Pull request by tessi**
there is a lot of duplicated code between the timelog and reports controller. DRY that.

### CODE REVIEW

**Comment by tessi** [comment]
seems to make things better just by moving them in a common helper: https://codeclimate.com/repos/52330269f3ea006d790604fa/compare/feature%2Frefactor_timelog_and_reports_controller

File: app/helpers/timelog_helper.rb [hunk 1]
```diff
@@ -212,4 +212,63 @@ def polymorphic_new_time_entry_path(object)
...
+    @from, @to = @to, @from if @from && @to && @from > @to
+    @from ||= (TimeEntry.earliest_date_for_project(@project) || Date.today)
+    @to   ||= (TimeEntry.latest_date_for_project(@project) || Date.today)
+  end
+
+  def find_optional_project
+    if !params[:issue_id].blank?
+      @issue = WorkPackage.find(params[:issue_id])
+      @project = @issue.project
+    elsif !params[:work_package_id].blank?
```
Comment (mfrister) [inline app/helpers/timelog_helper.rb:266]: Yay for issue_id and work_package_id. I assume there's a reason for that.

**Comment by mfrister** [comment]
:+1:

File: app/helpers/timelog_helper.rb (hunk 1 above)
Comment (tessi) [inline app/helpers/timelog_helper.rb:266]: Thought the same :) However, I was frightened to remove the `issue_id` thingy

**Pull request merged**